### ISSUE TITLE
CLI option to write to file

### ISSUE THREAD

**Issue by b00ga** [hadolint/hadolint#863]
- [ ] This is a bug report
- [X] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
Would like to write output to a file from hadolint command directly (not via shell redirection).

### Actual behavior
hadolint only writes results to STDOUT and shell redirection must  be used if output-to-file is desired.

### Steps to reproduce the behavior
tl;dr

I'd like an option similar to --output= or --output-file= that allows me to specify a file instead of STDOUT, so my CI can read that file for reporting results in its UI.

Gorey details:

Using JetBrains TeamCity for my CI. TC has Docker features that report on what docker containers were used/pulled in a build. It also has a feature that can read certain report types and interpret them for viewing in the web UI.

I'm trying to use the [Docker Wrapper feature](https://www.jetbrains.com/help/teamcity/docker-wrapper.html), which runs a command inside a Docker container. I'm using the hadolint/hadolint:latest container from dockerhub. The TC feature looks to be exec'ing a shell script wrapper they developed to call Docker, so I have no control of STDIN or STDOUT. The build jobs workdir is passed in as a volume, so I can get at the intended Dockerfile as input easily.

Since I don't have access to STOUT or a shell with this wrapper feature, there's no way to capture the output to provide to the checkstyle report processor. If hadolint had a file output option, I could write the output to a r/w docker volume so the TC agent (aka the docker host) could consume this file after `docker run` completes.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.10.0
```

**Comment by m-ildefons** [comment]
Hi Shawn,
a similar feature was requested not too long ago in  #837. I wonder however, if you would be able to workaround this using not the bare-bones `hadolint/hadolint:latest` container image but rather the `hadolint/hadolint-debian` or `hadolint/hadolint-alpine` image. Those two come with a shell and using [a custom entrypoint](https://www.jetbrains.com/help/teamcity/docker-wrapper.html#Setting+Image+Entrypoint), you should be able to use that shell to redirect Hadolint's output to your volume.
That being said, I'm currently working on the file-output feature. It still needs quite a bit cleanup before I can submit a PR, but it'll probably come.
